src/frida/repl.py

```diff
@@ -423,6 +431,57 @@ def _create_repl_script(self):
                 with codecs.open(self._user_script, 'rb', 'utf-8') as f:
                     user_script = f.read().rstrip("\r\n") + "\n\n// Frida REPL script:\n"
 
+            if self._codeshare_uri is not None:
+                trust_store = self._get_or_create_truststore()
+                project_url = 'https://codeshare.frida.re/api/project/{}/'.format(self._codeshare_uri)
+                response_json = None
+                try:
+                    response = urlopen(project_url)
+                    response_content = response.read()
+                    response_json = json.loads(response_content)
+                except Exception as e:
+                    self._print("Got an unhandled exception while trying to retrieve {} - {}".format(self._codeshare_uri, e))
+
+                if response_json:
+                    trusted_signature = trust_store.get(self._codeshare_uri, '')
+                    fingerprint = hashlib.sha256(response_json['source'].encode('utf-8')).hexdigest()
+                    if fingerprint == trusted_signature:
+                        user_script = response_json['source']
+                    else:
+                        self._print("""Hello! This is the first time you're running this particular snippit, or the snippit's source code has changed.
+
+Project Name: {project_name}
+Author: {author}
+Slug: {slug}
+Fingerprint: {fingerprint}
+URL: {url}
+                        """.format(
+                            project_name=response_json['project_name'],
+                            author="@" + self._codeshare_uri.split('/')[0],
+                            slug=self._codeshare_uri,
+                            fingerprint=fingerprint,
+                            url="https://codeshare.frida.re/@{}".format(self._codeshare_uri)
+                        ))
+
+                        while True:
+                            input_string = "Are you sure you'd like to trust this project? [y/N] "
+                            try:
+                                response = raw_input(input_string)
+                            except NameError:
+                                response = input(input_string)
+
+                            if response.lower() in 'no' or response == '':
```

Comment:
That's fair, I figured this was a more succinct way to do it, but you're right that it'll accept a few keys as y/n, I just figured no-one would ever hit those corner cases :-P. Happy to change it to be more explicit though! :)